scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -152,3 +154,42 @@ def __iter__(self):
 
     def get_generated_keys(self) -> List[str]:
         return [self._name]
+
+
+class AttackPointIteratorRepeat(AttackPointIterator):
+    """
+    Attack point iterator class that iterates 
+    over a configuration a repeat amount of times.
+    """
+
+    def __init__(self,
+                 operation: str,
+                 configuration: Dict,
+                 repetitions: int = 0) -> None:
+        """Initialize the repeated iterate if repetitions is not present
+          or set to 0 it will not repeat and if it set 
+          to a negative number it will do an infinite loop."""
+        assert "repeat" == operation
+        self._configuration_iterator = build_attack_points_iterator(
+            configuration)
+        if repetitions > 0:
+            self._repetitions = repetitions
+            self._len = repetitions * len(self._configuration_iterator)
+        elif repetitions == 0:
```

Comment:
This is redundant (result is the same as if you had the first condition `if repetitions >= 0:` (ok here you save one call to `len`).